### DISCUSSION TITLE
Rules defining

### DISCUSSION THREAD

**Discussion by Raks2110** [jertel/elastalert2#210] [Q&A] [answered]
This is my rules.yaml file 

data:
  elastalert.yaml: |-
    name: check this out
    type: frequency
    index: cloud-*
    num_events: 2
    timeframe:
        minutes: 1
    filter:
    - query:
        query_string:
            query: "bad AND certificate"
            default_field: log
    - query:
        query_string:
            query: "unable AND to AND get AND control AND connection AND for AND device"
            default_field: log
    - query:
        query_string:
            query: "Stream AND is AND nil"
            default_field: log
    - query:
        query_string:
            query: "certificate AND has AND expired AND or AND is AND not AND yet AND valid"
            default_field: log
    alert:
    - "googlechat"
    googlechat_webhook_url: "***********"
    alert_text: |
        Failure point-log: {0}
        Timestamp: {2}
        Pod: {1}
        Namespace: {3}
        Container: {4}
    alert_text_type: alert_text_only
    alert_text_args: ["log","Pod","@timestamp","Namespace","Container_ID"]    

The trigger is not fired on on google hangout , i seee this log multiple times "certificate has expired or is not yet valid"

And my log on kibana is 

*1|ERROR|2021/06/02 08:12:24.289688|warpClient: Handshake: [some ip] x509: certificate has expired or is not yet valid: current time 2021-06-02T08:12:24Z is after 2020-10-16T16:50:46Z

**Comment by jertel** [comment, marked as answer]
I suggest enabling debug on ElastAlert 2, this will give you more information on what's going on. Also remove all queries except for the one you know should match, to simplify your rule while you troubleshoot. Once you get it working you can slowly add back more complexity to your rule to find out what's breaking it.

**Reply by Raks2110** [reply]
if i just use one query it is working , how to add multiple rules

**Reply by jertel** [reply]
The documentation covers the subject of writing filters: https://elastalert2.readthedocs.io/en/latest/recipes/writing_filters.html#writingfilters, and advises you to review Elastic and Lucene documentation for more detailed information.

**Reply by Raks2110** [reply]
I could not find any work around for this , more help from anyone around here will be great ..

**Reply by nsano-rururu** [reply]
>if i just use one query it is working , how to add multiple rules

I thought it would be good to separate the rule files and write one query condition for each, but do you recognize it?

**Reply by nsano-rururu** [reply]
>I could not find any work around for this , more help from anyone around here will be great ..

Isn't it the idea that the question isn't answered well if you don't get an answer?

**Reply by Raks2110** [reply]
You mean to write different yaml files for each query ??

**Reply by Raks2110** [reply]
this is in kubernetes environment , i am using configmaps ,,,should i use multiple configmaps or multiple rule files inside a single configmap

**Reply by nsano-rururu** [reply]
>this is in kubernetes environment , i am using configmaps ,,,should i use multiple configmaps or multiple rule files inside a single configmap

I don't know that. Be clear what you want to hear.
I don't know about kubernetes at all.

Rather than setting the rule file individually to configmap, it is an image of setting the rule directory to configmap and starting the elastalert pod with k8s. Is it different?

Mounting the rules directory should be the same for docker, so I think it's the same for k8s.

**Reply by nsano-rururu** [reply]
>You mean to write different yaml files for each query ??

Then what do you want to do?

**Reply by nsano-rururu** [reply]
I don't want to separate files even though I say multiple rules. I do not understand

**Reply by Raks2110** [reply]
apiVersion: v1
kind: ConfigMap
metadata:
  name: logstash-rule
  namespace: 
  labels:
    k8s-app: elastalert
data:
  elastalert.yaml: |-
    name: check this out
    type: frequency
    index: cloud-*
    num_events: 2
    timeframe:
        minutes: 1
    filter:
    - query:
        query_string:
            query: "INFO"
            default_field: log
    alert:
    - "googlechat"
    googlechat_webhook_url: 
    alert_text: |
        Failure point-log: {0}
        Timestamp: {2}
        Pod: {1}
        Namespace: {3}
        Container: {4}
    alert_text_type: alert_text_only
    alert_text_args: ["log","Pod","@timestamp","Namespace","Container_ID"]
  elastalert-TEST.yaml: |- 
    name: check this out
    type: frequency
    index: cloud-*
    num_events: 2
    timeframe:
        minutes: 1
    filter:
    - query:
        query_string:
            query: "ERROR"
            default_field: log
    alert:
    - "googlechat"
    googlechat_webhook
    alert_text: |
        Failure point-log: {0}
        Timestamp: {2}
        Pod: {1}
        Namespace: {3}
        Container: {4}
    alert_text_type: alert_text_only
    alert_text_args: ["log","Pod","@timestamp","Namespace","Container_ID"]

**Reply by Raks2110** [reply]
will this work , i have two data files for two queries ....

**Reply by nsano-rururu** [reply]
Isn't it possible to write the settings of multiple rule files like the following? .. I don't know

apiVersion: v1
kind: ConfigMap
metadata:
name: logstash-rule
namespace:
labels:
k8s-app: elastalert
data:
rule1.yaml: |-

～Rule 1 settings ～

rule2.yaml: |-

～Rule 2 settings ～

rule3.yaml: |-

～Rule 3 settings ～

**Reply by Raks2110** [reply]
i will try with this method and will check n update

**Reply by Raks2110** [reply]
The above format you suggested throws an error stating "Rule2.yaml" is duplicated
